### ISSUE TITLE
elasticsearch: elasticsearch1 module does not get patched

### ISSUE THREAD

**Issue by dan-blanchard** [DataDog/dd-trace-py#671]
We have a few services that use both the latest version of `elasticsearch` for communicating with our ES 6.x cluster, and `elasticsearch1` for communicating with our legacy 1.7 cluster. 

I recently noticed that we have no ES trace data for anything that talks to our 1.7 cluster, because `ddtrace` only patches the `elasticsearch` package and not `elasticsearch1` (or any of the other versioned ones they put out). It would be enormously helpful for us if it could be updated to patch `elasticsearch1` as well. 

There are also `elasticsearch2` and `elasticsearch5` packages that others might be using. They create a new version-suffixed package every time a new major version comes out.

**Comment by brettlangdon** [comment]
@dan-blanchard thank you so much for bringing to our attention. I had no idea there were `elasticsearchX` packages available!!

**Comment by brettlangdon** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [Add support for version specific elasticsearch packages](https://trello.com/c/hiqSaE0Q/1221-add-support-for-version-specific-elasticsearch-packages)

**Comment by dan-blanchard** [comment]
@brettlangdon we pretty desperately need this diagnose an issue we've got right now. Is there any hacky workaround you could recommend?

**Comment by brettlangdon** [comment]
@dan-blanchard I will try to dig a little deeper today, but I think the only "workaround" is to do `pip install elasticsearch==1.x.x` :/

The issue we have is that we explicitly `import elasticsearch` in our patching: https://github.com/DataDog/dd-trace-py/blob/7db12f1c398c07cd5baf91c571aed672dbb6496d/ddtrace/contrib/elasticsearch/patch.py#L1-L3

Another work around would be to fork this repo, make a copy of `ddtrace/contrib/elasticsearch` as `ddtrace/contrib/elasticsearch1` and update the imports and the `required_modules = ['elasticsearch']'` in `ddtrace/contrib/elasticsearch1/__init__.py`, then you can at least manually patch via `from ddtrace.contrib.elasticsearch1 import patch; patch()`


Please let me know if neither of those temporary options would work for you, I am happy to keep digging into possible solutions for you.

**Comment by brettlangdon** [comment]
@dan-blanchard I just opened up #701 which *should* work.

This is something I hacked together pretty quick, not sure when we will be able to release it, but if you wanted to test and report back on success/fail, you can install it via:

```
pip install git+https://github.com/dd-trace-py@brettlangdon/all.es.packages
```

The TL;DR; it changes it so we patch the first module found of `elasticsearch{,1,2,5}`

**Comment by dan-blanchard** [comment]
Thanks, but we actually need to trace both `elasticsearch` and `elasticsearch1` in our code (since we're talking to both ES1 and 6 clusters), so just grabbing the first one like #701 does won't really work for us.

**Comment by brettlangdon** [comment]
ok, I think I can find a work around for that, but I might not get to it by EOD today :(

**Comment by dan-blanchard** [comment]
I appreciate the help. No worries about not having it done by EOD! We're going to try copy-pasting out the `patch` function, modifying it to use `elasticsearch1` explicitly, and then calling it directly. Something like #701 (that works for all the modules) is obviously a better long-term solution, but we'll see what we can make work in the short term.

**Comment by brettlangdon** [comment]
For an idea of what I am thinking. Add helper functions for `patch` and `unpatch` but which take a module, then the real `patch` and `unpatch` will just do a loop (similar to what I have in `/elasticsearch.py`) and patch/unpatch all available modules.

**Comment by brettlangdon** [comment]
@dan-blanchard give that branch a try now, it should patch all available modules now, instead of the first one.

**Comment by brettlangdon** [comment]
Support for `elasticsearch{1,2,5}` packages has been released with version `0.17.0` of `ddtrace`.
